Reword beginners' course lessons -- first batch (#546)
pyvec/naucse.python.cz: lessons/beginners/and-or/index.md

    </tr>
    <tr>
        <td><code>and</code></td>
        <td><code>True and False</code><br><code>2 < 3 and 5 < 3</code></td>
        <td>„a zároveň“</td>
    </tr>
    <tr>
        <td><code>or</code></td>
        <td><code>True or False</code><br><code>2 < 3 or 5 < 3</code></td>
        <td>„a nebo“</td>
    </tr>
    <tr>
        <td><code>not</code></td>
        <td><code>not False</code><br><code>not 5 < 3</code</td>
        <td>„ne“</td>
    </tr>
</table>

Například, chceš-li zjistit, jestli je kterékoli z dvou čísel záporné,
můžeš napsat:

```python
a = float(input("Zadej první stranu obdélníka: ")
b = float(input("Zadej druhou stranu obdélníka: ")

if a <= 0 or b <= 0:
    print("Délka nemůže být záporná!")
```

> > [warning] Falešní kamarádi

Reviewer: tenhle výraz znám z výuky angličtiny (actually vs. aktuálně - false friends), ale v tomto kontextu mu úplně nerozumím. Budou mu rozumět čtenáři?

Author: Já považuju `and` a `or` za hodně podobný případ jako “false friends” v angličtině: `if a or b <= 0` neznamená „jestli je a nebo b menší než nula“.

Myslím, že když tomu někdo rozumět nebude, tak se nic moc nestane. Je to jen nadpis. Ale pro toho, kdo ten termín zná, to může být přínosná paralela.

>

> > Pozor na to, že `and` a `or` nejsou anglická slovíčka, ale operátory

Reviewer: ```suggestion
> Pozor na to, že `and` a `or` nejsou anglická slovíčka, ale operátory,
```

> které spojují logické výrazy.

> > Na *obě* strany `and` i `or` patří porovnání (nebo proměnná/výraz,

Reviewer: ```suggestion
> Na *obě* strany `and` i `or` patří výraz, jehož hodnota je `True`/`False`
```

> >jehož hodnota je `True`/`False`):

Reviewer: ```suggestion
> (například porovnání).
```

>
> ```python
> if a <= 0 or b <= 0:
> ```
>
> Může se zdát, že by se to dalo zkrátit a napsat `if a or b <= 0:` – „pokud
> je A nebo B menší než 0“.
> Ale to si Python přeloží na:
>
> ```python
> if (a) or (b <= 0):
> ```
>
> ... tedy  „pokud platí A, a nebo je B menší než 0“.
> A to moc smyslu nedává.
> (Kdy „platí“ celé číslo?)

## Šťastná/Bohatá

Pro příklad použijeme `and` ve vylepšeném programu, který rozdává nejapné rady
do života.
Zkus si ho projít a okomentovat části, které nejsou na první pohled jasné.

```python
# Tento program rozdává naivní rady do života.

print('Odpovídej "ano" nebo "ne".')
stastna_retezec = input('Jsi šťastná? ')
if stastna_retezec == 'ano' or stastna_retezec == 'Ano':
    stastna = True
elif stastna_retezec == 'ne' or stastna_retezec == 'Ne':
    stastna = False
else:

>     print('Nerozumím!')

Reviewer: později přijde `NameError`, ale to se tady nezměnilo

bohata_retezec = input('Jsi bohatá? ')

> if bohata_retezec == 'ano' or stastna_retezec == 'Ano':

Reviewer: ```suggestion
if bohata_retezec == 'ano' or bohata_retezec == 'Ano':
```

    bohata = True

> elif bohata_retezec == 'ne' or stastna_retezec == 'Ne':

Reviewer: ```suggestion
elif bohata_retezec == 'ne' or bohata_retezec == 'Ne':
```
